### PR TITLE
chore: Adds `mongodbatlas_encryption_at_rest_private_endpoint` acceptance test using azapi to approve private endpoint & check ACTIVE status

### PR DESCRIPTION
**Pull request by maastha**
## Description

Adds `mongodbatlas_encryption_at_rest_private_endpoint` acceptance test using azapi to approve private endpoint & check ACTIVE status and minor improvements to tests.

The new test uses azure/azapi Terraform provider which can log sensitive information in CI like Azure subscriptionID used in parent_id of the resource in case of failures, hence this test is disabled to run in CI but maybe run locally without any additional setup.

Link to any related issue(s): [CLOUDP-270777](https://jira.mongodb.org/browse/CLOUDP-270777)

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR. A migration guide must be created or updated if the new feature will go in a major version.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR. A migration guide must be created or updated.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contributing guides](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/README.md)
- [ ] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [ ] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [ ] I have added any necessary documentation (if appropriate)
- [ ] I have run make fmt and formatted my code
- [ ] If changes include deprecations or removals I have added appropriate changelog entries.
- [ ] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

File: internal/service/encryptionatrest/resource_migration_test.go [hunk 1]
```diff
@@ -37,7 +38,7 @@ func TestMigEncryptionAtRest_basicAWS(t *testing.T) {
 		Steps: []resource.TestStep{
 			{
 				ExternalProviders: mig.ExternalProviders(),
-				Config:            testAccMongoDBAtlasEncryptionAtRestConfigAwsKms(projectID, &awsKms, false), // not using data source as it was introduced in 1.19.0
+				Config:            testAccMongoDBAtlasEncryptionAtRestConfigAwsKms(projectID, &awsKms, useDatasource), // not using data source as it was introduced in 1.19.0
```
Comment (lantoli) [inline internal/service/encryptionatrest/resource_migration_test.go]: knit: you may want to do some opportunistic improvements, like renaming `testAccMongoDBAtlasEncryptionAtRestConfigAwsKms` to `configAwsKms', etc.

File: internal/service/encryptionatrest/resource_migration_test.go [hunk 2]
```diff
@@ -134,6 +126,8 @@ func TestMigEncryptionAtRest_basicAzure(t *testing.T) {
 			"subscription_id":     azureKeyVault.GetSubscriptionID(),
 			"tenant_id":           azureKeyVault.GetTenantID(),
 		}
+
+		useDatasource = mig.IsProviderVersionAtLeast("1.19.0") // data source introduced in this version
```
Comment (lantoli) [inline internal/service/encryptionatrest/resource_migration_test.go:130]: thanks for following this up

File: internal/service/encryptionatrest/resource_migration_test.go [hunk 3]
```diff
@@ -142,23 +136,14 @@ func TestMigEncryptionAtRest_basicAzure(t *testing.T) {
 		Steps: []resource.TestStep{
 			{
 				ExternalProviders: mig.ExternalProviders(),
-				Config:            acc.ConfigEARAzureKeyVault(projectID, &azureKeyVault, false, false), // not using data source as it was introduced in 1.19.0
+				Config:            acc.ConfigEARAzureKeyVault(projectID, &azureKeyVault, false, useDatasource), // not using data source as it was introduced in 1.19.0
```
Comment (lantoli) [inline internal/service/encryptionatrest/resource_migration_test.go]: remove comment as info is already in the useDatasource var assignment

File: internal/service/encryptionatrestprivateendpoint/resource_test.go [hunk 4]
```diff
@@ -70,6 +71,53 @@ func basicTestCase(tb testing.TB) *resource.TestCase {
 	}
 }
 
+func TestAccEncryptionAtRestPrivateEndpoint_approveEndpointWithAzureProvider(t *testing.T) {
+	acc.SkipTestForCI(t) // uses azure/azapi Terraform provider which can log sensitive information in CI like Azure subscriptionID used in parent_id of the resource
```
Comment (lantoli) [inline internal/service/encryptionatrestprivateendpoint/resource_test.go:75]: if we know the patterns, GH allows to hide sensitive info, e.g.: https://github.com/marketplace/actions/hide-sensitive-inputs

but maybe it's not possible in this situation

File: internal/service/encryptionatrestprivateendpoint/resource_test.go [hunk 5]
```diff
@@ -70,6 +71,53 @@ func basicTestCase(tb testing.TB) *resource.TestCase {
...
+	)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { acc.PreCheckBasic(t); acc.PreCheckEncryptionAtRestEnvAzure(t); acc.PreCheckPreviewFlag(t) },
+		ExternalProviders:        acc.ExternalProvidersOnlyAzapi(),
+		ProtoV6ProviderFactories: acc.TestAccProviderV6Factories,
+		CheckDestroy:             checkDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: configPrivateEndpointAzureBasic(projectID, azureKeyVault, region, false),
```
Comment (lantoli) [inline internal/service/encryptionatrestprivateendpoint/resource_test.go]: knit: similar comment, rename `configPrivateEndpointAzureBasic`, `checkPrivateEndpointAzureBasic` to something like `configAzureBasic`, `checkAzureBasic`...

**Review by lantoli — APPROVED** [review]
knit: PR description checklists are not filled

File: internal/service/encryptionatrestprivateendpoint/resource_test.go (hunk 4 above)
Comment (maastha) [reply internal/service/encryptionatrestprivateendpoint/resource_test.go:75]: The issue is we don't need to hide once specific attribute here that is a direct input to the action. The attribute that's concerning is the parent_id which is constructed/initialized inside the TF config. 
Then when a failure happens, the azure provider includes that value in it's failure message. So essentially it's the failure message that can log that information.

Does that make sense?

File: internal/service/encryptionatrest/resource_migration_test.go (hunk 1 above)
Comment (maastha) [reply internal/service/encryptionatrest/resource_migration_test.go]: updated, thanks for the tip!

File: internal/service/encryptionatrest/resource_migration_test.go (hunk 3 above)
Comment (maastha) [reply internal/service/encryptionatrest/resource_migration_test.go]: updated

File: internal/service/encryptionatrestprivateendpoint/resource_test.go (hunk 5 above)
Comment (maastha) [reply internal/service/encryptionatrestprivateendpoint/resource_test.go]: updated

**Pull request merged**